### PR TITLE
Set 'number_of_samples' in constructor of ADCBase and subclasses

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1393) |

#### Main changes
- Set 'number_of_samples' in constructor of ADCBase and subclasses
- Add range of returned value in SoundSensor

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: packages/pma/pitop/pma/adc_base.py [hunk 1]
```diff
@@ -38,7 +44,7 @@ def own_state(self):
         }
 
     def read(
-        self, number_of_samples=1, delay_between_samples=0.05, peak_detection=False
+        self, delay_between_samples=0.05, peak_detection=False, number_of_samples=None
```
Comment (angusjfw) [inline packages/pma/pitop/pma/adc_base.py:47]: is there a reason for reordering these params? i think that can be considered a breaking change in python since keyword params can still be used positionally?

File: packages/pma/pitop/pma/light_sensor.py [hunk 2]
```diff
@@ -15,8 +15,16 @@ class LightSensor(ADCBase):
     :param str port_name: The ID for the port to which this component is connected
     """
 
-    def __init__(self, port_name, pin_number=1, name="light_sensor"):
-        ADCBase.__init__(self, port_name=port_name, pin_number=pin_number, name=name)
+    def __init__(
+        self, port_name, pin_number=1, name="light_sensor", number_of_samples=3
```
Comment (angusjfw) [inline packages/pma/pitop/pma/light_sensor.py:19]: i think we should mention in the doc description that by default 3 samples are taken so the reading takes around 0.5 seconds

File: packages/pma/pitop/pma/adc_base.py (hunk 1 above)
Comment (jcapona) [reply packages/pma/pitop/pma/adc_base.py]: my bad!

File: packages/pma/pitop/pma/light_sensor.py (hunk 2 above)
Comment (jcapona) [reply packages/pma/pitop/pma/light_sensor.py:24]: makes sense; also just added documentation for the `name` parameter which was missing

**Review by angusjfw — APPROVED** [review]
